src/index.js

```diff
@@ -489,7 +515,33 @@ class InputElement extends React.Component {
     }
   }
 
+  snapCursorToLastChar = () => {
+    if (this.isCursorPastLastChar()) {
+      this.setCursorToEnd();
+    }
+  }
+
+  ensureCursorSnapsToLastChar = (eventType) => {
```

Comment:
Keyboard events have a lot of issues on older mobile platforms and current code doesn't handle some edge cases like pasting text on a phone. I think it's better to modify `saveSelectionLoop` to call `beforeSaveSelection` method and place cursor adjustments there. Doing that we won't need to rely on event listeners.